ui/src/components/query/builder/edit/dialogs/SaveCustomSetDialog.vue

```diff
@@ -0,0 +1,229 @@
+import { useToast } from "primevue/usetoast";
+import * as yup from "yup";
+
+interface Props {
+  setMembers: Node[];
+}
+const props = defineProps<Props>();
+
+const schema = yup.object({
+  scheme: yup.string().required().label("Scheme").default(IM.NAMESPACE),
+  iri: yup
+    .string()
+    .required()
+    .test("isValidIri", "Iri exists", async () => await isValidIri())
+    .label("Iri"),
+  name: yup.string().required().label("Name"),
+  type: yup.string().required().label("Type").default(IM.CONCEPT_SET)
+});
+
+const toast = useToast();
+
+const { defineComponentBinds, handleSubmit, resetForm, errors, setFieldValue } = useForm({
+  validationSchema: schema
+});
+
+const fullIri: ComputedRef<string> = computed(() => `${scheme.value.modelValue || ""}${iri.value.modelValue || ""}`);
+
+const scheme = defineComponentBinds("scheme");
+const iri = defineComponentBinds("iri");
+const name = defineComponentBinds("name");
+const type = defineComponentBinds("type");
+
+const schemeOptions: Ref<TTIriRef[]> = ref([]);
+const typeOptions: Ref<TTIriRef[]> = ref([]);
+const selectedMember: Ref<Node> = ref({});
+const showSaveCustomSetDialog = ref(false);
+
+watch(
+  () => showSaveCustomSetDialog.value,
+  async () => await init()
+);
+
+onMounted(async () => {
+  await init();
+});
+
+async function init() {
+  typeOptions.value = await getTypeOptions();
+  schemeOptions.value = await getSchemeOptions();
+  selectDefaults();
+}
+
+async function isValidIri() {
+  if (!scheme.value.modelValue) return false;
+  const exists = await EntityService.iriExists(fullIri.value);
+  return !exists;
+}
+
+async function getTypeOptions(): Promise<TTIriRef[]> {
+  const setTypes = await EntityService.getEntityChildren(IM.SET);
+  return setTypes.map(setType => {
+    return {
+      "@id": setType["@id"],
+      name: setType.name
+    };
+  });
+}
+
+async function getSchemeOptions(): Promise<TTIriRef[]> {
+  const schemes = await EntityService.getEntityChildren(IM.NAMESPACE + "Graph");
+  return schemes.map(scheme => {
+    return {
+      "@id": scheme["@id"],
+      name: scheme.name
+    };
+  });
+}
+
+function onNameGenIri() {
+  setFieldValue("iri", name.value.modelValue);
+}
+
+function selectDefaults() {
+  setFieldValue("scheme", IM.NAMESPACE);
+  setFieldValue("type", IM.CONCEPT_SET);
+}
+
+const onSubmit = handleSubmit(async () => {
+  const setEntity = buildSetEntity();
+  try {
+    await FilerService.fileEntity(setEntity, IM.NAMESPACE, IM.ADD_QUADS);
+    const createdEntity = await EntityService.getFullEntity(setEntity["@id"]);
+    if (isObjectHasKeys(createdEntity, [RDFS.LABEL, RDF.TYPE, IM.HAS_STATUS, IM.SCHEME, IM.IS_CONTAINED_IN, IM.DEFINITION]))
+      toast.add({ severity: "success", summary: "Created", detail: 'Created "' + createdEntity[RDFS.LABEL], life: 3000 });
+  } catch (e: any) {
+    toast.add({ severity: "error", summary: e.response.data.title, detail: e.response.data.detail, life: 3000 });
```

Comment:
why catch error and not let the error propagate through the error handler?

Reply:
That would go to a 500 page losing all progress in the query builder. Better to catch the error here and not let the app fail.